client/src/main/scala/com.gu.contentapi.client/ContentApiClient.scala

```diff
@@ -44,9 +50,42 @@ trait ContentApiClient {
   /** Authentication and format parameters appended to each query */
   private def parameters = Map("api-key" -> apiKey, "format" -> "thrift")
 
+  val maxRetries: Int = 3 // max wait time ~1.5s
+
   /** Streamlines the handling of a valid CAPI response */
-  private def fetchResponse(contentApiQuery: ContentApiQuery)(implicit context: ExecutionContext): Future[Array[Byte]] =
-    get(url(contentApiQuery), headers).flatMap(HttpResponse.check)
+  private def fetchResponse(contentApiQuery: ContentApiQuery, attempt: Int = 1)(implicit context: ExecutionContext): Future[Array[Byte]] =
+    fetchResponseWithBackoff(contentApiQuery, attempt)(context).flatMap { _.flatMap { HttpResponse.check }}
+
+  private def fetchResponseWithBackoff(contentApiQuery: ContentApiQuery, attempt: Int)(implicit context: ExecutionContext): CancellableFuture[Future[HttpResponse]] = {
+    val waitDuration = scheduledExecutor.getWaitDuration(attempt)
+    // To test this - uncomment the following line
+    // println(s"fetchResponseWithBackoff attempt: ${attempt} waitDuration: ${waitDuration}")
+    // and add 400 to the HttpResponse failedButMaybeRecoverable set
+    // and something like 400 -> "Not a real recoverable - just for testing" to the recoverableErrorMessages list, then
+    // run the tests, and the should "handle error responses" test will cause these printlns to be output:
+    //    fetchResponseWithBackoff attempt: 1 waitDuration: 0 milliseconds
+    //    fetchResponseWithBackoff attempt: 2 waitDuration: 500 milliseconds
+    //    fetchResponseWithBackoff attempt: 3 waitDuration: 1000 milliseconds
+    // and ultimately fail with the message
+    //    [info] - should handle error responses *** FAILED ***
+    //    [info]   The future returned an exception of type: org.scalatest.exceptions.TestFailedException, with message: com.gu.contentapi.client.model.ContentApiRecoverableException: Not a real recoverable - just for testing was not equal to com.gu.contentapi.client.model.ContentApiError: Bad Request. (GuardianContentClientTest.scala:53)
+
+    val op = get(url(contentApiQuery), headers)(context)
```

Comment:
This will start the operation, as `Future` is not pure, so the `delayExecution` will not work as you expect. The only way to get around that is to pass it in the form of a lazy parameter, so:
```scala
trait ScheduledExecutor { 
  def delayExecution[A](duration: Duration)(f: ExecutionContext => Future[A]): Future[A]
}

delayExecution(waitDuration) { implicit ec =>
  get(url(contentApiQuery), headers)
}
```

You need to pass the `ExecutionContext` otherwise the future will run on the scheduled executor's thread pool 😨